### PR TITLE
don't allow empty string team

### LINKED ISSUE #8095: agent need to acquire in their team
related to #8074 I do not understand the reason for having team="" go to *any* team. This is highly un-desirable IMO and WQE should just get in nowhere land. Better the request should not be able to assign if Team="" is the one that is decided.

I don't know why this is happening more frequently now, but this is happening and impacting urgent requests

### PR DESCRIPTION
**Pull request by ticoann**
fixes #8095

### CODE REVIEW

File: src/couchapps/WorkQueue/lists/workRestrictions.js [hunk 1]
```diff
@@ -55,7 +55,11 @@ function(head, req) {
         };
 		
         var ele = row["doc"]["WMCore.WorkQueue.DataStructs.WorkQueueElement.WorkQueueElement"];
-
+        // team shouldn't be empty string or undefined
+        if (teams.length === 0 || !ele["TeamName"]) {
```
Comment (amaltaro) [inline src/couchapps/WorkQueue/lists/workRestrictions.js]: The line below is performing almost the same checks. How about we either add `|| teams.indexOf(ele["TeamName"]) === -1` to this if and remove the if below or keep the if
below with only the indexOf check?

**Review by amaltaro — COMMENTED** [review]
Functionality looks good. Made a comment in the code for an improvement though.
Perhaps you could also update the PR title to something like 'Don't pull work if team name is an empty string'

**Comment by amaltaro** [comment]
Seangchan, is this the fix you applied to the hepcloud and relval team? I'm updating the twiki we forget it. Once this patch is updated, we should also apply this patch to the other relval agent (down atmo).

File: src/couchapps/WorkQueue/lists/workRestrictions.js (hunk 1 above)
Comment (ticoann) [reply src/couchapps/WorkQueue/lists/workRestrictions.js]: Yes you are right! I can modify that but I need to check why all the unittest failed.

**Comment by ticoann** [comment]
Unfortunately, this patch need to apply in GQ not just LQ. With LQ application it just prevent job to be created but not prevent pulling the job. I will do some damage control. I will stop the WorkQueueManagers in relval and hepcould for a while.

**Comment by amaltaro** [comment]
Seangchan, as we quickly discussed, how about you also change this code:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WorkQueue/WorkQueueReqMgrInterface.py#L215

such that it considers only non-empty strings? Something like `request['Teams'] = [t for t in request['Teams'] if t]`

**Comment by ticoann** [comment]
closing it replaced by #8108

**Pull request closed without merging**